ros2param/test/test_cli_delete.py

```diff
@@ -0,0 +1,188 @@
+            assert param_command.wait_for_shutdown(timeout=5)
+        assert param_command.exit_code == launch_testing.asserts.EXIT_OK
+        assert launch_testing.tools.expect_output(
+            expected_lines=['Deleted parameter successfully'],
+            text=param_command.output,
+            strict=True
+        )
+        with self.launch_param_command(
+                arguments=['get', '/param_delete_node', 'int_param']) as param_command:
+            assert param_command.wait_for_shutdown(timeout=5)
+        assert param_command.exit_code == launch_testing.asserts.EXIT_OK
+        assert launch_testing.tools.expect_output(
+            expected_lines=['Parameter not set.'],
+            text=param_command.output,
+            strict=True
+        )
+
+    @launch_testing.markers.retry_on_failure(times=3)
+    def test_delete_int_param_hidden_node(self):
+        with self.launch_param_command(
+                arguments=['delete', '/_hidden_param_delete_node', 'int_param']) as param_command:
+            assert param_command.wait_for_shutdown(timeout=5)
+        assert param_command.exit_code == 1
+        assert launch_testing.tools.expect_output(
+            expected_text='Node not found\n',
+            text=param_command.output,
+            strict=True
+        )
+
+    @launch_testing.markers.retry_on_failure(times=5)
+    def test_delete_int_param_hidden_node_with_hidden_flag(self):
+        with self.launch_param_command(
+                arguments=[
+                    'delete', '--include-hidden-nodes',
+                    '/_hidden_param_delete_node', 'int_param']) as param_command:
+            assert param_command.wait_for_shutdown(timeout=8)
+        assert param_command.exit_code == launch_testing.asserts.EXIT_OK
+        assert launch_testing.tools.expect_output(
+            expected_lines=['Deleted parameter successfully'],
+            text=param_command.output,
+            strict=True
+        )
+        with self.launch_param_command(
+                arguments=[
+                    'get', '--include-hidden-nodes',
+                    '/_hidden_param_delete_node', 'int_param']) as param_command:
+            assert param_command.wait_for_shutdown(timeout=5)
+        assert param_command.exit_code == launch_testing.asserts.EXIT_OK
+        assert launch_testing.tools.expect_output(
+            expected_lines=['Parameter not set.'],
+            text=param_command.output,
+            strict=True
+        )
+
+    @launch_testing.markers.retry_on_failure(times=3)
+    def test_delete_any_param_nonexistent_node(self):
+        with self.launch_param_command(
+                arguments=['delete', '/foo/nonexistent_node', 'int_param']) as param_command:
+            assert param_command.wait_for_shutdown(timeout=5)
+        assert param_command.exit_code == 1
+        assert launch_testing.tools.expect_output(
+            expected_text='Node not found\n',
```

Comment:
@BMarchi consider using `expected_lines=['Node not found']` instead (better not use newline literals).